cmd/kubeadm/app/discovery/discovery.go

```diff
@@ -48,7 +48,7 @@ func For(cfg *kubeadmapi.JoinConfiguration) (*clientcmdapi.Config, error) {
 	clusterinfo := kubeconfigutil.GetClusterFromKubeConfig(config)
 	return kubeconfigutil.CreateWithToken(
 		clusterinfo.Server,
-		cfg.ClusterName,
+		"cluster-discovery",
```

Comment:
shouldn't we be getting the clustername from the cluster here?

Reply:
I don't see a way to do this at this point, do we want to call here `kubeconfigutil.ToClientSet` and then `configutil.FetchConfigFromFileOrCluster` in order to call `kubeconfigutil.CreateWithToken` with the correct cluster name?

Reply:
@neolit123 I have added a separate commit addressing your comment, fetching the cluster name from the context, I don't know if it's the best approach.